### PR TITLE
BFB-500: Refactor Ring Behaviour

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-500

**Description**
Change all at once behaviour to use a queue and consolidate into one script (cause all the behaviour is about the same now anyways)

### CODE REVIEW

File: Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs [hunk 1]
```diff
@@ -42,9 +42,9 @@ private void CreateSystems()
             {
                 RingSystem newSystem = grouping.RingSystemType switch
                 {
-                    RingSystemTypes.Standard => new OrderedRingSystem(grouping.Rings),
-                    RingSystemTypes.DisplayAllAtOnce => new AllAtOnceRingSystem(grouping.Rings),
-                    RingSystemTypes.RandomSelection => new OrderedRingSystem(grouping.Rings, true),
+                    RingSystemTypes.Standard => new RingSystem(grouping.Rings),
+                    RingSystemTypes.DisplayAllAtOnce => new RingSystem(grouping.Rings, allAtOnce: true),
+                    RingSystemTypes.RandomSelection => new RingSystem(grouping.Rings, isRandomized: true),
```
Comment (m4rtint) [inline Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs:47]: @Gursaanj are we still doing random? 
or are we doing predesigned only?

**Comment by m4rtint** [comment]
So this change makes it so 1 ring shows up one at a time. 
Are we not doing all at the same time + billboarding numbers? 

or is that for another ticket? 

@Gursaanj

**Comment by unoctium1** [comment]
@m4rtint I only changed the behaviour of the all at once ring system. Now they all show up at once still, but must be completed in order. Standard and random are unchanged

File: Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs (hunk 1 above)
Comment (Gursaanj) [reply Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs:47]: We will be doing preassigned but I say lets just keep this in for now, just so we can pla

> So this change makes it so 1 ring shows up one at a time. Are we not doing all at the same time + billboarding numbers?
> 
> or is that for another ticket?
> 
> @Gursaanj

The Billboarding will be another ticket!

Comment (Gursaanj) [reply Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs:47]: We are only doing preassigned, so I think it is safe to delete the random selection

Comment (unoctium1) [reply Assets/_BForBoss/_Core/Scripts/Labor/RingSystem/RingLaborManager.cs:47]: I removed the random selection from the scene - Now the current demo loop is 5 all at once, 2 all at once, and then 5 in order sequential

**Comment by m4rtint** [comment]
> @m4rtint I only changed the behaviour of the all at once ring system. Now they all show up at once still, but must be completed in order. Standard and random are unchanged

i meant like remove the randomized code. 

But that's fine for now.

**Review by m4rtint — APPROVED** [review]

**Pull request merged**